src/framework/components/animation/component.js

```diff
@@ -458,7 +458,14 @@ class AnimationComponent extends Component {
 
     onBeforeRemove() {
         for (var i = 0; i < this.assets.length; i++) {
-            var asset = this.system.app.assets.get(this.assets[i]);
+
+            // this.assets can be an array of pc.Assets or an array of numbers (assetIds)
+            var assetId = this.assets[i];
+            if (typeof assetId !== 'number') {

```

Comment:
I would do `        if (value instanceof Asset) {` here instead. Kinda like:

https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/component.js#L394

This seems to be the way the check is done throughout the framework (seach for the string `instanceof Asset`).